doctests

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+"""
+A shell-executable script to run all the commands contained in README.md or specified in EXTRA_TESTS and
+assert that Wheatley does not crash.  With decent test coverage, this prevents mistakes like silly crashes
+and argument errors.
+"""
+
+from subprocess import STDOUT, check_output, CalledProcessError, TimeoutExpired
+import shlex
+
+# ID of a tower that I made called 'DO NOT ENTER'
+ROOM_ID = "238915467"
+EXAMPLE_METHOD = "Plain Bob Major"
+IGNORE_STRING = "<!--- doctest-ignore -->"
+
+
+# Tests that should be run on top of the examples from README.md
+EXTRA_TESTS = []
+
+
+def get_all_tests():
+    """ Get all commands that should be tested (these should start with 'wheatley [ID NUMBER]'). """
+
+    tests = []
+
+    with open("README.md") as readme:
+        is_in_code = False
+        ignore_next_examples = False
+
+        for i, l in enumerate(readme.read().split("\n")):
+            stripped_line = l.strip()
+
+            # If we see the line IGNORE_STRING, we should ignore the next block of examples
+            if not is_in_code and stripped_line == IGNORE_STRING:
+                ignore_next_examples = True
+
+            if stripped_line.startswith("```"):
+                # Clear the ignore flag whenever we *finish* a code block
+                if is_in_code:
+                    ignore_next_examples = False
+
+                # Alternate into or out of code blocks
+                is_in_code = not is_in_code
```

Comment:
I think this would flow slightly better the other way around
```suggestion
            if stripped_line.startswith("```"):
                # Alternate into or out of code blocks
                is_in_code = not is_in_code
                
                # Clear the ignore flag whenever we *finish* a code block
                if not is_in_code:
                    ignore_next_examples = False
```

Reply:
Yep that looks good!